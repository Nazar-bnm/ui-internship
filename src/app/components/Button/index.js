import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  return (
    <button className="button">
      {props.icon && <i className={props.icon + ' icon'}></i>}
      {props.children}
    </button>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  children: React.PropTypes.oneOfType([
    React.PropTypes.arrayOf(React.PropTypes.node),
    React.PropTypes.node,
  ]).isRequired,
};

export default Button;
