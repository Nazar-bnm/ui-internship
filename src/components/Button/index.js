import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  return (
    <button className="button">
      {props.icon && <i className={props.icon + ' icon'}></i>}
      {props.children}
    </button>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Button;
