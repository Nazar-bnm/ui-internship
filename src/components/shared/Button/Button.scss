@import '../../../scss/global_styles';

.go-back-btn {
  width: 80px;
  height: 30px;
  margin: 10px;
}