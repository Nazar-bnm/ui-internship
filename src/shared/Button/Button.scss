@import '../../scss/global_styles';

.go-back-btn {
  width: 80px;
  height: 30px;
  margin: 10px;
}

.login-register__button {
  height: 30px;
  width: 160px;
  margin-right: 10px;
  padding: 0 5px;
  background-color: $black;
  border: none;
  color: $white;
  font-size: 0.6rem;
  text-transform: uppercase;

  &:hover {
    transform: scale(1.1);
  }
}

@media only screen and (min-width: 480px) {
  .login-register__button {
    width: 110px;
  }
}