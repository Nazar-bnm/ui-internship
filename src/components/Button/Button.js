import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = (props) => {
  const {
    customClass, icon, children, onClickFunction
  } = props;

  return (
    <button
      className={`${customClass} button`}
      type="button"
      onClick={onClickFunction}
    >
      {icon && <i className={`${icon} icon`} />}
      {children}
    </button>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  customClass: PropTypes.string,
  onClickFunction: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

Button.defaultProps = {
  icon: null,
  customClass: null,
  onClickFunction: null,
  children: []
};

export default Button;
