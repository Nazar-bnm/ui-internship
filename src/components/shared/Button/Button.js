import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './Button.scss';

const Button = (props) => {
  const {
    className, icon, children, onClick
  } = props;

  const CN = 'button';

  return (
    <button
      className={cx(CN, className)}
      type="button"
      onClick={onClick}
    >
      {icon && <i className={`${icon} icon`} />}
      {children}
    </button>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

Button.defaultProps = {
  icon: null,
  className: '',
  children: []
};

export default Button;
