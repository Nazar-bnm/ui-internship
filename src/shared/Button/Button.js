import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = (props) => {
  const {
    className, icon, children, onClick
  } = props;

  return (
    <button
      className={`${className} button`}
      type="button"
      onClick={onClick}
    >
      {icon && <i className={`${icon} icon`} />}
      {children}
    </button>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

Button.defaultProps = {
  icon: null,
  className: null,
  onClick: () => {},
  children: []
};

export default Button;
