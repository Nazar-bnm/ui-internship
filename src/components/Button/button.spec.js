import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';

describe('<Button />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Button />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
