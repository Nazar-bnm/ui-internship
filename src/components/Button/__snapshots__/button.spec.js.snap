// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  className="null button"
  onClick={null}
  type="button"
/>
`;
